kotest-assertions/kotest-assertions-shared/src/jvmMain/kotlin/io/kotest/assertions/counter.kt

```diff
@@ -5,11 +5,11 @@ actual val assertionCounter: AssertionCounter = ThreadLocalAssertionCounter
 
 object ThreadLocalAssertionCounter : AssertionCounter {
 
-   private val context = object : ThreadLocal<Int>() {
+   val values = object : ThreadLocal<Int>() {
       override fun initialValue(): Int = 0
    }
 
-   override fun get(): Int = context.get()
-   override fun reset() = context.set(0)
-   override fun inc() = context.set(context.get() + 1)
+   override fun get(): Int = values.get()
+   override fun reset() = values.set(0)
+   override fun inc() = values.set(values.get() + 1)
```

Comment:
Is this safe if multiple coroutines are running for the same test simultaneously?

If all we need to track is "an assertion did / did not occur", perhaps an atomic boolean would be easier to reason about?